./app.py:17:1 'utils.utils.sum_dicts' imported but unused
./worker_functions.py:1:1 'json.dumps' imported but unused
./models/discussion.py:4:1 'mongoengine.PULL' imported but unused
./models/discussion.py:9:1 'mongoengine.fields.DictField' imported but unused
./models/discussion.py:9:1 'mongoengine.fields.ReferenceField' imported but unused
./managers/discussion_manager.py:9:1 'typing.Any' imported but unused
./managers/discussion_manager.py:9:1 'typing.Dict' imported but unused
./managers/discussion_manager.py:9:1 'typing.List' imported but unused
./managers/discussion_manager.py:9:1 'typing.Optional' imported but unused
./managers/discussion_manager.py:9:1 'typing.Tuple' imported but unused
./managers/discussion_manager.py:19:1 'utils.utils' imported but unused
./managers/discussion_manager.py:447:9 local variable 'discussion' is assigned to but never used
./managers/discussion_manager.py:505:9 local variable 'discussion' is assigned to but never used
./managers/discussion_manager.py:620:9 local variable 'discussion' is assigned to but never used
./managers/board_manager.py:1:1 'datetime' imported but unused
./managers/board_manager.py:2:1 'logging' imported but unused
./managers/board_manager.py:3:1 'typing.Any' imported but unused
./managers/board_manager.py:3:1 'typing.Dict' imported but unused
./managers/board_manager.py:3:1 'typing.List' imported but unused
./managers/board_manager.py:3:1 'typing.Optional' imported but unused
./managers/board_manager.py:3:1 'typing.Tuple' imported but unused
./managers/board_manager.py:11:1 'constants' imported but unused
./managers/board_manager.py:12:1 'utils.utils' imported but unused
./managers/board_manager.py:14:1 'models.discussion.User' imported but unused
./managers/tests/discussion_manager_test.py:1:1 'asyncio' imported but unused
./managers/tests/discussion_manager_test.py:3:1 'time' imported but unused
./managers/tests/discussion_manager_test.py:244:9 local variable 'unit_id1' is assigned to but never used
./managers/tests/discussion_manager_test.py:249:9 local variable 'unit_id2' is assigned to but never used
./managers/tests/board_manager_test.py:3:1 'uuid' imported but unused
./managers/tests/board_manager_test.py:6:1 'models.discussion.Discussion' imported but unused
worker_functions.py:2: error: Cannot find implementation or library stub for module named 'managers.global_manager'
tests/app_test.py:3: error: Skipping analyzing 'socketio': found module but no type hints or library stubs
tests/app_test.py:3: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
worker.py:4: error: Module 'worker_functions' has no attribute 'expire_discussion'
worker.py:20: error: Incompatible types in assignment (expression has type "Callable[[Any], Any]", base class "Worker" defined the type as "Optional[StartupShutdown]")
worker.py:21: error: Incompatible types in assignment (expression has type "Callable[[Any], Any]", base class "Worker" defined the type as "Optional[StartupShutdown]")
worker.py:22: error: Incompatible types in assignment (expression has type "List[Any]", base class "Worker" defined the type as "Dict[str, Union[Function, CronJob]]")
app.py:7: error: Skipping analyzing 'jsonschema': found module but no type hints or library stubs
app.py:9: error: Skipping analyzing 'socketio': found module but no type hints or library stubs
app.py:13: error: Cannot find implementation or library stub for module named 'managers.global_manager'
app.py:14: error: Cannot find implementation or library stub for module named 'schema.board_responses'
app.py:14: error: Cannot find implementation or library stub for module named 'schema'
app.py:15: error: Cannot find implementation or library stub for module named 'schema.discussion_requests'
app.py:16: error: Cannot find implementation or library stub for module named 'schema.discussion_responses'
app.py:17: error: Cannot find implementation or library stub for module named 'utils.utils'
Found 14 errors in 4 files (checked 7 source files)
INFO:engineio.server:Server initialized for aiohttp.
DEBUG:asyncio:Using selector: EpollSelector
DEBUG:aioredis:Creating tcp connection to ('redis', 6379)
.
----------------------------------------------------------------------
Ran 1 test in 0.034s

OK
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-3' coro=<RedisConnection._read_data() running at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:186> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f9d0ebc5fa0>()]> cb=[RedisConnection.__init__.<locals>.<lambda>() at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:168]>
INFO:engineio.server:Server initialized for aiohttp.
DEBUG:asyncio:Using selector: EpollSelector
DEBUG:aioredis:Creating tcp connection to ('redis', 6379)
INFO:root:<function nothing at 0x7f61ce3de0d0> DISC ERR
.INFO:engineio.server:Server initialized for aiohttp.
DEBUG:aioredis:Creating tcp connection to ('redis', 6379)
.INFO:engineio.server:Server initialized for aiohttp.
DEBUG:aioredis:Creating tcp connection to ('redis', 6379)
.INFO:engineio.server:Server initialized for aiohttp.
DEBUG:aioredis:Creating tcp connection to ('redis', 6379)
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-9' coro=<RedisConnection._read_data() done, defined at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:180> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f61c07602b0>()]> cb=[RedisConnection.__init__.<locals>.<lambda>() at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:168]>
.INFO:engineio.server:Server initialized for aiohttp.
DEBUG:aioredis:Creating tcp connection to ('redis', 6379)
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-3' coro=<RedisConnection._read_data() running at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:186> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f61c91c4790>()]> cb=[RedisConnection.__init__.<locals>.<lambda>() at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:168]>
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-6' coro=<RedisConnection._read_data() running at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:186> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f61c0753610>()]> cb=[RedisConnection.__init__.<locals>.<lambda>() at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:168]>
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-12' coro=<RedisConnection._read_data() done, defined at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:180> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f61c075a8e0>()]> cb=[RedisConnection.__init__.<locals>.<lambda>() at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:168]>
.INFO:engineio.server:Server initialized for aiohttp.
DEBUG:aioredis:Creating tcp connection to ('redis', 6379)
.INFO:engineio.server:Server initialized for aiohttp.
DEBUG:aioredis:Creating tcp connection to ('redis', 6379)
.INFO:engineio.server:Server initialized for aiohttp.
DEBUG:aioredis:Creating tcp connection to ('redis', 6379)
.
----------------------------------------------------------------------
Ran 8 tests in 0.418s

OK
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-15' coro=<RedisConnection._read_data() running at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:186> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f61c06fbbe0>()]> cb=[RedisConnection.__init__.<locals>.<lambda>() at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:168]>
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-24' coro=<RedisConnection._read_data() done, defined at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:180> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f61c0705a60>()]> cb=[RedisConnection.__init__.<locals>.<lambda>() at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:168]>
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-18' coro=<RedisConnection._read_data() running at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:186> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f61c073f940>()]> cb=[RedisConnection.__init__.<locals>.<lambda>() at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:168]>
ERROR:asyncio:Task was destroyed but it is pending!
task: <Task pending name='Task-21' coro=<RedisConnection._read_data() running at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:186> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f61c0705790>()]> cb=[RedisConnection.__init__.<locals>.<lambda>() at /usr/local/lib/python3.8/dist-packages/aioredis/connection.py:168]>
